commit 746fc76b820dce8cbb17a1e5e70a1558db4d7406
Merge: 328141e51e6f 8fcdc238ce1b
Author: Linus Torvalds <torvalds@linux-foundation.org>
Date:   Thu Aug 4 19:47:37 2022 -0700

    Merge tag 'scsi-misc' of git://git.kernel.org/pub/scm/linux/kernel/git/jejb/scsi
    
    Pull SCSI updates from James Bottomley:
     "Updates to the usual drivers (ufs, qla2xx, target, lpfc, smartpqi,
      mpi3mr).
    
      The main driver change that might cause issues on down the road is the
      conversion of some of our oldest surviving drivers to the DMA API
      (should only affect m68k).
    
      The only major core change is the rework of async resume; the rest are
      either completely trivial or for updating deprecated APIs"
    
    * tag 'scsi-misc' of git://git.kernel.org/pub/scm/linux/kernel/git/jejb/scsi: (195 commits)
      scsi: target: Remove XDWRITEREAD emulated support
      scsi: megaraid: Remove the static variable initialisation
      scsi: ch: Do not initialise statics to 0
      scsi: ufs: core: Fix spelling mistake "Cannnot" -> "Cannot"
      scsi: target: iscsi: Do not require target authentication
      scsi: target: iscsi: Allow AuthMethod=None
      scsi: target: iscsi: Support base64 in CHAP
      scsi: target: iscsi: Add support for extended CDB AHS
      scsi: ufs: dt-bindings: Add SC8280XP binding
      scsi: target: iscsi: Fix clang -Wformat warnings
      scsi: ufs: core: Read device property for ref clock
      scsi: libsas: Resume SAS host for phy reset or enable via sysfs
      scsi: hisi_sas: Modify v3 HW SATA completion error processing
      scsi: hisi_sas: Relocate DMA unmap of SMP task
      scsi: hisi_sas: Remove unnecessary variable to hold DMA map elements
      scsi: hisi_sas: Call hisi_sas_slave_configure() from slave_configure_v3_hw()
      scsi: mpi3mr: Delete a stray tab
      scsi: mpi3mr: Unlock on error path
      scsi: mpi3mr: Reduce VD queue depth on detecting throttling
      scsi: mpi3mr: Resource Based Metering
      ...

diff --cc drivers/ufs/host/ufshcd-pltfrm.c
index 173aea8e9997,bde34685eb7b..5739ff007828
--- a/drivers/ufs/host/ufshcd-pltfrm.c
+++ b/drivers/ufs/host/ufshcd-pltfrm.c
@@@ -108,20 -108,9 +108,20 @@@ out
  	return ret;
  }
  
 +static bool phandle_exists(const struct device_node *np,
 +			   const char *phandle_name, int index)
 +{
 +	struct device_node *parse_np = of_parse_phandle(np, phandle_name, index);
 +
 +	if (parse_np)
 +		of_node_put(parse_np);
 +
 +	return parse_np != NULL;
 +}
 +
  #define MAX_PROP_SIZE 32
- static int ufshcd_populate_vreg(struct device *dev, const char *name,
- 				struct ufs_vreg **out_vreg)
+ int ufshcd_populate_vreg(struct device *dev, const char *name,
+ 			 struct ufs_vreg **out_vreg)
  {
  	char prop_name[MAX_PROP_SIZE];
  	struct ufs_vreg *vreg = NULL;